crates/common/src/utils.rs

```diff
@@ -89,3 +79,31 @@ pub fn check_l2_range_exists<DB: SharedLedgerOps>(
     }
     false
 }
+
+pub fn extract_sequencer_commitments<Da: DaService>(
+    sequencer_da_pub_key: &[u8],
+    l1_block_hash: [u8; 32],
+    da_data: &mut [<<Da as DaService>::Spec as DaSpec>::BlobTransaction],
+) -> Vec<SequencerCommitment> {
+    let mut sequencer_commitments = vec![];
+    // if we don't do this, the zk circuit can't read the sequencer commitments
+    da_data.iter_mut().for_each(|blob| {
+        blob.full_data();
+    });
+    da_data.iter_mut().for_each(|tx| {
+        let data = DaDataBatchProof::try_from_slice(tx.full_data());
+        // Check for commitment
+        if tx.sender().as_ref() == sequencer_da_pub_key {
+            if let Ok(DaDataBatchProof::SequencerCommitment(seq_com)) = data {
+                sequencer_commitments.push(seq_com);
+            } else {
```

Comment:
I don't think this else branch is needed. Because since this call is `extract_sequencer_commitment`, if it contains proofs, we don't really care about them. So the warn message is going to be misleading.

Reply:
We had it from before but the same concept applies.